src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1569,6 +1570,34 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Gets the request body decoded as array, typically from a JSON payload.
+     *
+     * @throws JsonException When the body cannot be decoded to an array
+     */
+    public function toArray(): array
+    {
+        if ('' === $content = $this->getContent()) {
+            throw new JsonException('Response body is empty.');
+        }
+
+        try {
+            $content = json_decode($content, true, 512, \JSON_BIGINT_AS_STRING | (\PHP_VERSION_ID >= 70300 ? \JSON_THROW_ON_ERROR : 0));
```

Comment:
Related to my previous comment, alternatively we could move the json error check (for PHP < 7.3) under this statement, but throwing the PHP json exception instead.
```php
\JsonException(json_last_error_msg(), json_last_error());
```